### PR TITLE
Send cache hit/miss events to statsd

### PR DESCRIPTION
**Pull request by jcristau**
Addresses #2766

### CODE REVIEW

**Comment by jcristau** [comment]
@jbuck does this look plausible?  Do we need to set different variables/prefixes for e.g. admin vs public, or are they going to end up separated on the way to grafana anyway?

**Review by bhearsum — APPROVED** [review]
This looks fine to me.

There's probably other things we should look at instrumenting as well -- I'm thinking of BadDataErrors, OutdatedDataError, and maybe some other things -- but that's probably best done as a follow-up after we know the basics work.

File: src/auslib/util/cache.py [hunk 1]
```diff
@@ -24,6 +25,7 @@ class MaybeCacher(object):
     def __init__(self):
         self.caches = {}
         self._make_copies = False
+        self.stats_client = StatsClient(prefix="balrog.cache")
```
[the remark this answers is not in the document]
Comment (jcristau) [reply src/auslib/util/cache.py:28]: haha yes it was nice to have that wrapper to hook into :)

[the remark this answers is not in the document]
Comment (jbuck) [reply src/auslib/util/cache.py:28]: Could you verify that I can configure the host/port via [environment variables ](https://statsd.readthedocs.io/en/v3.3/configure.html#from-the-environment)? Or add the plumbing to this instance of StatsClient?

**Comment by jbuck** [comment]
> @jbuck does this look plausible? Do we need to set different variables/prefixes for e.g. admin vs public, or are they going to end up separated on the way to grafana anyway?

The metrics sent by each instance of the balrog app will have the following tags applied to every statsd metric:

    app = "balrog"
    environment = "stage" # dev/stage/prod etc
    realm = "nonprod" # or prod

Adding an additional tag for admin vs public vs agent is probably the best way to tackle this, however it seems that python statsd client doesn't support that, so I guess a change in the field name/prefix/suffix is the way to go

**Review by gbrownmozilla — APPROVED** [review]

**Comment by jcristau** [comment]
https://github.com/mozilla-releng/balrog/pull/2798/commits/562a47d6bbd98c642e5660286861c28140bcb09e should address @jbuck's comments:
- uses settings from the environment
- sets default prefix to balrog.admin.cache or balrog.public.cache depending on the app.

**Review by gabrielBusta — APPROVED** [review]

**Pull request merged**